Source/WebCore/platform/graphics/Font.h

```diff
@@ -425,6 +429,52 @@ ALWAYS_INLINE FloatRect Font::boundsForGlyph(Glyph glyph) const
     return bounds;
 }
 
+ALWAYS_INLINE Vector<FloatRect> Font::boundsForGlyphs(std::span<const Glyph> glyphs) const
+{
+    Vector<FloatRect> glyphBounds(glyphs.size(), FloatRect());
+    Vector<Glyph> glyphsNeedingMeasurement;
+    Vector<size_t> positionsNeedingMeasurement;
+    glyphsNeedingMeasurement.reserveInitialCapacity(glyphs.size());
+    positionsNeedingMeasurement.reserveInitialCapacity(glyphs.size());
+
+    for (size_t glyphIndex = 0; glyphIndex < glyphs.size(); ++glyphIndex) {
+        auto glyph = glyphs[glyphIndex];
+        if (isZeroWidthSpaceGlyph(glyph))
+            continue;
+
+        if (m_glyphToBoundsMap) {
+            auto bounds = m_glyphToBoundsMap->metricsForGlyph(glyph);
+            if (bounds.width() != cGlyphSizeUnknown) {
+                glyphBounds[glyphIndex] = bounds;
+                continue;
+            }
+        }
+
+        glyphsNeedingMeasurement.append(glyph);
+        positionsNeedingMeasurement.append(glyphIndex);
+    }
+
+    if (glyphsNeedingMeasurement.isEmpty())
+        return glyphBounds;
+
+    if (!m_glyphToBoundsMap)
+        m_glyphToBoundsMap = makeUnique<GlyphMetricsMap<FloatRect>>();
+
+    const auto& measuredBounds = platformBoundsForGlyphs(glyphsNeedingMeasurement);
```

Comment:
I believe `platformBoundsForGlyphs` is defined only for CoreText but the caller here `boundsForGlyphs` is defined for all platforms.